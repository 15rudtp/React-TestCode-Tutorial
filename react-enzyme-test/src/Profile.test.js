import React from "react";
import {render} from '@testing-library/react';
import Profile from './Profile';

 /**
  * 랜더링 테스트
  */

describe('<Profile/>', () => {
    it('matches snapshot', () => {
        const utils = render(<Profile username="song" name="송경세"/>);
        expect(utils.container).toMatchSnapshot();
    })
    it('shows the props correctly', () => {
        const utils = render(<Profile username="song" name="송경세"/>);
        utils.getByText('song');
        utils.getByText('(송경세)');
        utils.getByText(/송/);
        
        //getByText: 컴포넌트에 해당 문자열이 존재하는지 검사.
    })
});