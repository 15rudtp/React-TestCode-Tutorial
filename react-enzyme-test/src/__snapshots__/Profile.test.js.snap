// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> matches snapshot 1`] = `
<div>
  <div>
    <b>
      song
    </b>
     
    <span>
      (
      송경세
      )
    </span>
  </div>
</div>
`;
